libp2p/protocols/connectivity/autonat/service.nim

```diff
@@ -153,7 +156,30 @@ proc schedule(service: AutonatService, switch: Switch, interval: Duration) {.asy
   heartbeat "Scheduling AutonatService run", interval:
     await service.run(switch)
 
+proc addressMapper(
+  self: AutonatService,
+  peerStore: PeerStore,
+  listenAddrs: seq[MultiAddress]): Future[seq[MultiAddress]] {.gcsafe, async.} =
+
+  var addrs = newSeq[MultiAddress]()
+  for listenAddr in listenAddrs:
+    var processedMA = listenAddr
+    try:
+      let hostIP = initTAddress(listenAddr).get()
+      if not hostIP.isGlobal():
+        if self.networkReachability == NetworkReachability.Reachable:
```

Comment:
Maybe this if should be the first thing we do? ie early return if we are not reachable

Reply:
This is the only thing which is done. I merged both ifs.